### ISSUE TITLE
Not working from inside a goroutine

### ISSUE THREAD

**Issue by hajsf** [getlantern/systray#195]
In my app, I'm trying to run 3 goroutines:
- http listner
- ui using [lorca](https://github.com/zserge/lorca)
- systray
Both `http listner` and `lorca ui` are working fine, but the `systray` is not executed, my code is:
```go
var ui lorca.UI
var err error

func main() {
	go func() {
	systray.Run(onReady, onExit)
	}()
	// Start Host goroutine
	go func() {

		http.Handle("/static/", http.StripPrefix("/static/", http.FileServer(http.Dir("./public"))))
		http.HandleFunc("/", index)
		http.ListenAndServe(":8090", nil)
	}()

	// Start UI
	ui, err = lorca.New("http://localhost:8090/", "", 1200, 800)
	if err != nil {
		log.Fatal(err)
	}
	defer ui.Close()
	ui.Bind("setLanguage", func(data string) {
		// perform the server side task here
		fmt.Println(data)
	})

	<-ui.Done()
}
```
If I tried to execute `systray.Run(onReady, onExit)` without goroutine, then the `systray` is working fine, but blocking the other 2 goroutines frombing executed!

I'm running my code at `MacBook Air macOS Big Sur Version 11.2.2`

**Comment by Altiano** [comment]
i have the same issue too

**Comment by mc51** [comment]
I'd be interested in a solution as well.

**Comment by tarasmatsyk** [comment]
This happens because of `runtime.LockOSThread()` in the init function which ensures that current goroutine fully occupies the current thread and does not "let in" other goroutines. I don't know why author has chosen to run it on the module init, probably to make the module safe and simple. 

However, there is a workaround, you can release the current thread and occupy it (hopefully another one) in a different goroutine. Here is an example:
```
package main

import (
	"runtime"

	"github.com/getlantern/systray"
	"github.com/getlantern/systray/example/icon"
	"golang.org/x/sync/errgroup"
)

func main() {
	var g errgroup.Group
	runtime.UnlockOSThread() // release thread
	g.Go(func() error {
		runtime.LockOSThread() // reserve it again, systray will release it on Quit
		systray.Run(onReady, onExit)
		return nil
	})
	err := g.Wait()
	if err != nil {
		panic(err)
	}
}

func onReady() {
	systray.SetIcon(icon.Data)
	systray.SetTitle("Awesome App")
	systray.SetTooltip("Pretty awesome超级棒")
	mQuit := systray.AddMenuItem("Quit", "Quit the whole app")

	// Sets the icon of a menu item. Only available on Mac and Windows.
	mQuit.SetIcon(icon.Data)
}

func onExit() {
	// clean up here
}

```

Native libs in Go is quite new to me, so I cannot sure my fix does not have any negative implications. What I am not sure about is what happens when everything runs in one thread which is reserved by a sub-goroutine. There are could be other things I am not aware of.

**Comment by tarasmatsyk** [comment]
This happens because of `runtime.LockOSThread()` in the init function which ensures that current goroutine fully occupies the current thread and does not "let in" other goroutines. I don't know why author has chosen to run it on the module init, probably to make the module safer and simpler. 

However, there is a workaround, you can release the current thread and occupy it (hopefully another one) in a different goroutine. Here is an example:
```
package main

import (
	"runtime"
	"time"

	"github.com/getlantern/systray"
	"github.com/getlantern/systray/example/icon"
	"golang.org/x/sync/errgroup"
	"golang.org/x/sys/unix"
)

func main() {
	var g errgroup.Group
	runtime.UnlockOSThread()
	g.Go(func() error {
		for range time.Tick(2 * time.Second) {
			println(unix.Getpid())
		}
		return nil
	})
	g.Go(func() error {
		runtime.LockOSThread()
		println(unix.Getpid())
		systray.Run(onReady, onExit)
		return nil
	})
	err := g.Wait()
	if err != nil {
		panic(err)
	}
}

func onReady() {
	systray.SetIcon(icon.Data)
	systray.SetTitle("Awesome App")
	systray.SetTooltip("Pretty awesome超级棒")
	mQuit := systray.AddMenuItem("Quit", "Quit the whole app")

	// Sets the icon of a menu item. Only available on Mac and Windows.
	mQuit.SetIcon(icon.Data)
}

func onExit() {
	// clean up here
}

```

Native libs in Go is quite new to me, so I cannot sure my fix does not have any negative implications. What I am not sure about is what happens when everything runs in one thread which is reserved by a sub-goroutine. There are could be other things I am not aware of.

**Comment by mc51** [comment]
Thanks for the idea @tarasmatsyk. I've been trying some things as well. However, I'm really at the beginning of understanding Go so things are not clear yet. For Linux and Windows I've managed to run systray + another GUI Lib (in my case [goey](https://pkg.go.dev/bitbucket.org/rj/goey)). But I've found that MacOS seems to be special in its requirements regarding UI. It requires UI stuff to be running on the main thread only. Otherwise, the app will crash similar to this:
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'NSWindow drag regions should only be invalidated on the Main Thread!'

I guess, this is why systray locks itself in the init function. At the same time, GUI libs usually try do to the same (e.g. goey does this).  So far, I don't see any solution on MacOS for running systray + another UI. It seems that they must both be running on the main thread and that seems hard (if not imposssible?). 

I'm not sure about @hajsf using Lorca though. Since (I guess) Lorca is using chromium and not its own UI there might not be a problem: Start Lorca and your server on different threads and let systray run on the main thread.
